codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/customizations/GlacierAccountIdMiddlewareTest.kt

```diff
@@ -0,0 +1,69 @@
+import software.amazon.smithy.aws.traits.auth.UnsignedPayloadTrait
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.MemberShape
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareRenderableExecutionContext
+
+class GlacierAccountIdMiddlewareTest {
+    @Test
+    fun testGlacierMiddlewareRendersCorrectly() {
+        val writer = SwiftWriter("testName")
+        val serviceShape = ServiceShape.builder()
+            .id("com.test#Glacier")
+            .version("1.0")
+            .build()
+        val accountIdMember = MemberShape.builder()
+            .id("com.test#TestInputShapeName\$accountId")
+            .target("smithy.api#String")
+            .build()
+        val inputShape = StructureShape.builder()
+            .id("com.test#TestInputShapeName")
+            .addMember(accountIdMember)
+            .build()
+        val outputShape = StructureShape.builder().id("com.test#TestOutputShapeName").build()
+        val errorShape = StructureShape.builder().id("com.test#TestErrorShapeName").build()
+        val operationShape = OperationShape.builder()
+            .id("com.test#ExampleOperation")
+            .addTrait(UnsignedPayloadTrait())
+            .input { ShapeId.from("com.test#TestInputShapeName") }
+            .output { ShapeId.from("com.test#TestOutputShapeName") }
+            .addError("com.test#TestErrorShapeName")
+            .build()
+        val model = Model.builder()
+            .addShape(serviceShape)
+            .addShape(operationShape)
+            .addShape(accountIdMember)
+            .addShape(inputShape)
+            .addShape(outputShape)
+            .addShape(errorShape)
+            .build()
+        val context = model.newTestContext(serviceShapeId = "com.test#Glacier", generator = AWSRestJson1ProtocolGenerator()).ctx
+        val opStackName = "stack"
+        val glacierMiddleware = GlacierAccountIdMiddleware()
+
+        glacierMiddleware.render(model, context.symbolProvider, writer, operationShape, opStackName, MiddlewareRenderableExecutionContext.CLIENT)
+
+        val contents = writer.toString()
+        val expectedContents = """
+stack.initializeStep.intercept(position: .after, id: "GlacierAccountIdAutoFill") { (context, input, next) -> Swift.Result<ClientRuntime.OperationOutput<TestOutputShapeName>, ClientRuntime.SdkError<ExampleOperationOutputError>> in
+    guard let accountId = input.accountId, !accountId.isEmpty {
+        var copiedInput = input
+        copiedInput.accountId = "-"
+        return next.handle(context: context, input: copiedInput)
+    }
+    return next.handle(context: context, input: input)
+}"""
```

Comment:
I'm suprised that we didn't need to use .trimIndent() here, but okay, sure it works!